- добавлены Single Image View, zoom scale

headlinerten opened the pull request:
- добавлен zoom scale;
- переверстан экран профиля кодом.

headlinerten committed:
- добавлен Aingle Image View;
- добавлен zoom scale;
- переверстан экран профиля кодом.

So1ovyov commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -1,11 +1,111 @@
 import UIKit
 
+import UIKit
```
Дублируется импорт `UIKit`, давай удалим эту строку.

So1ovyov commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -1,11 +1,111 @@
+    // MARK: - Setup Methods
+    
+    private func setupUI() {
+        view.addSubview(profileImageView)
```
Обрати внимание на инструмент `forEach` при верстке и настройке элементов, он поможет тебе сократить повторяющийся код
Вот пример:
```
        [view1, 
         view2,
         view3].forEach {
            $0.translatesAutoresizingMaskIntoConstraints = false
             view.addSubview($0)
        }
```
Таким образом все строки `translatesAutoresizingMaskIntoConstraints` и `addSubview` можно удалить и сократить код

So1ovyov commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -1,11 +1,111 @@
+    
+    private let logoutButton: UIButton = {
+        let image = UIImage(systemName: "ipad.and.arrow.forward")
+        let button = UIButton.systemButton(with: image!, target: nil, action: nil)
```
Старайся избегать `force unwrap`, `force cast`, `force try`, даже если уверен, что значение будет. Лучше использовать опционал и конструкции для безопасного разворачивания: `guard let`, `if let`
```swift
        let image = UIImage(systemName: "ipad.and.arrow.forward")
        let button = UIButton.systemButton(
            with: image ?? UIImage(),
            target: nil,
            action: nil
        )
```

So1ovyov commented on ImageFeed/SingleImageView/SingleImageViewController.swift:
```diff
@@ -0,0 +1,78 @@
+    }
+    func scrollViewDidEndZooming(_ scrollView: UIScrollView, with view: UIView?, atScale scale: CGFloat) {
+            centerImageInScrollView()
+        }
```
Не забывай следить за отступами, общепринятые отступы помогают читать код быстрее. В общем случае поможет выделить весь файл `cmd+A` и зажать `control+i`

So1ovyov commented on ImageFeed/SingleImageView/SingleImageViewController.swift:
```diff
@@ -0,0 +1,78 @@
+    func viewForZooming(in scrollView: UIScrollView) -> UIView? {
+        imageView
+    }
+    func scrollViewDidEndZooming(_ scrollView: UIScrollView, with view: UIView?, atScale scale: CGFloat) {
```
Предлагаю методы разделять между собой одной пустой строкой, чтобы было легче воспринимать код и где начинается/заканчивается каждая сущность

headlinerten committed:
- Учтены замечания из PR;
- исправлена четность расстановки лайков;
- изменен цвет стрелки "Назад" в navigationBar с голубого на белый.

So1ovyov commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -1,39 +1,58 @@
         super.viewDidLoad()
-
+        
         tableView.rowHeight = 200
```
В данной настройке нет необходимости, так как высоту ячейки ты рассчитываешь ниже в методе `heightForRowAt`. Так что строку `tableView.rowHeight = 200` можно удалить.

So1ovyov commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -1,39 +1,58 @@
-
+    
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: ImagesListCell.reuseIdentifier, for: indexPath)
```
Обрати внимание, в кейсах с `guard` можно совершать сразу несколько действий, например создавать ячейку и кастить её:
```
        guard let imageListCell = tableView.dequeueReusableCell(
            withIdentifier: ImagesListCell.reuseIdentifier,
            for: indexPath
        ) as? ImagesListCell else {
            return UITableViewCell()
        }
```

So1ovyov commented on ImageFeed/Profile/ProfileViewController.swift:
```diff
@@ -1,11 +1,111 @@
+    
+    private func setupUI() {
+        [profileImageView, nameLabel, usernameLabel, bioLabel, logoutButton].forEach {
+            $0.translatesAutoresizingMaskIntoConstraints = false
```
Отлично! Но забыл удалить все эти повторяющиеся строки выше, они там не нужны теперь.

So1ovyov commented:
@headlinerten критических замечаний не осталось, отличная работа.
Успехов в следующем спринте!
